governance/xc_admin/packages/xc_admin_common/src/contracts/config.ts

```diff
@@ -0,0 +1,58 @@
+import { ethers } from "ethers";
+import PythAbi from "@pythnetwork/pyth-sdk-solidity/abis/IPyth.json";
+import { Contract, ContractType, NetworkId } from "./Contract";
+import { EvmPythUpgradable } from "./EvmPythUpgradable";
+import { EvmWormholeReceiver } from "./EvmWormholeReceiver";
+
+export function getEvmProvider(
+  networkId: NetworkId,
+  networksConfig: any
+): ethers.providers.Provider {
+  const networkConfig = networksConfig["evm"][networkId]!;
+  return ethers.getDefaultProvider(networkConfig.url);
+}
+
+export function loadContractConfig(
+  contractsConfig: any,
+  networksConfig: any
+): Contract<any>[] {
+  const contracts = [];
+  for (const contractConfig of contractsConfig) {
+    contracts.push(fromConfig(contractConfig, networksConfig));
+  }
+  return contracts;
+}
+
+function fromConfig(contractConfig: any, networksConfig: any): Contract<any> {
+  switch (contractConfig.type) {
+    case ContractType.EvmPythUpgradable: {
+      const ethersContract = new ethers.Contract(
+        contractConfig.address,
+        PythAbi,
+        getEvmProvider(contractConfig.networkId, networksConfig)
+      );
+
+      return new EvmPythUpgradable(
+        contractConfig.networkId,
+        contractConfig.address,
+        ethersContract
+      );
```

Comment:
It's easier to understand (and less coupled) if this be a factory method in the contract file. And in general we automatically call the factory methods without having switch case at all.